packages/swing-store/src/transcriptStore.js

```diff
@@ -1,6 +1,7 @@
 // @ts-check
 import { Readable } from 'stream';
 import { Buffer } from 'buffer';
+import { gzipSync, gunzipSync } from 'zlib';
```

Comment:
A useful data point might be the typical cumulative size of the transcript items in existing on-chain transcript spans.  In particular, it's worth looking at how many cranks a typical span encompasses and what the average size of those is.  My intuition is that a typical span is a few hundred cranks and that they are mostly small, so the whole string for the span wouldn't be a major issue.  I'm quite sure we do have some spans whose concatenated item strings would be really big, but this would be due to individual large items (such as contract installs or vat starts) such that the size burden wouldn't be mitigated by streaming since the database API compels us to read the individual items synchronously anyway.

Reply:
currently a span is 200 crank, which is arguably low for some, and large for others given it's not linked to the amount of work or syscall performed. While I agree there is only at most a single very large delivery, it's possible for a vat to have a mean delivery that is large, in which case streaming would roughly divide memory impact by 200.

Reply:
I don't see where the factor of 200 comes from.  If you have 199 cranks that are 20 bytes and 1 crank that's five megabytes, you're still going to need five megabytes regardless of streaming or not.

Reply:
We cannot rule out 200 cranks of 1MB each, which if we're lucky compresses to 100KB. A non streaming version would allocate 200 MB in a string array, before joining them into a single 200MB string, and then compressing them into a 20MB buffer (likely with smaller buffers being accumulated first). A streaming version would feed the 1MB strings one at a time, and slowly build up an array of buffers totaling 20MB, before concatenating those into a single buffer.

Reply:
While we can't categorically rule out 200 successive cranks of 1MB each, I really don't think that's a realistic thing to worry about.  Things that generate transcript items that big tend to be rare, very high overhead operations, so if we're getting 200 of those in a row there's something seriously haywire and we're probably fighting bigger fires someplace else in the system.

Reply:
I've added #8739 to track this aspiration.